src/artifactRegistry.ts

```diff
@@ -45,6 +56,84 @@ export class ArtifactRegistryDockerRegistryClient {
     this.repositoryFields = { project, location, repository };
   }
 
+  async getGitCommitsBetweenTags({
+    prevTag,
+    nextTag,
+    dockerImageRepository,
+  }: {
+    prevTag: string;
+    nextTag: string;
+    dockerImageRepository: string;
+  }): Promise<string[]> {
+    core.info(
+      `running diff docker tags ${prevTag} ${nextTag} ${dockerImageRepository}`,
+    );
+    // Input is relatively trusted; this is largely to prevent mistaken uses
+    // like specifying a full Docker-style repository with slashes.
+    if (dockerImageRepository.includes('/')) {
+      throw Error('repository cannot contain a slash');
+    }
+
+    const dockerInfos = (
+      await this.client.listTags({
+        parent: this.client.pathTemplates.packagePathTemplate.render({
+          ...this.repositoryFields,
+          package: dockerImageRepository,
+        }),
+      })
+    )[0];
+
+    // We want to get the minimum tag for each version, since this implies those commits
+    // made a change to the docker image so are relevant to the diff.
+    const tagBoundsMap = new Map<string, { tag: string; commit: string }>();
+    for (const dockerInfo of dockerInfos) {
+      if (!dockerInfo.version || !dockerInfo.name) continue;
+
+      const tag = this.client.pathTemplates.tagPathTemplate.match(
+        dockerInfo.name,
+      ).tag;
+
+      // If it is a number we can ignore the tag
+      if (typeof tag !== 'string') continue;
+
+      // We only care about the tags between prev and next that have a git commit
+      const gitCommitMatches = tag.match(/-g([0-9a-fA-F]+)$/);
+      if (
+        ((tag >= prevTag && tag <= nextTag) ||
```

Comment:
At some level it should refuse to opine about the list of changes unless both given tags contain --- with the same string before it. Otherwise the comparison is invalid